pyccel/codegen/python_wrapper.py

```diff
@@ -160,7 +164,7 @@ def create_shared_library(codegen,
 
     # get the include folder path and library files
     recompile_object(cwrapper_lib,
-                      compiler = wrapper_compiler,
+                      compiler = c_compiler,
                       verbose  = verbose)
```

Comment:
In `pipeline.py` we have the line
```python
wrapper_compiler = Compiler('GNU', 'c', debug)
```
so `wrapper_compiler` is always the GCC compiler. Here you swap it with the C compiler selected by the user, but not so on lines 213-215. Why is that?

Reply:
Sorry this is probably a half-baked solution (it is still in draft). We use GNU to be compatible with Python (we should probably find a way to work out which compiler Python was actually compiled with). However I'm not sure if the functions that are called from the wrapper need to be compiled. I think we are doing because several of these functions are Python functions. However I am not sure that it is actually necessary.
I will probably try using the compiler's C compiler for everything

Reply:
> I will probably try using the compiler's C compiler for everything

This seems to work so I think I will stick with this solution